pcl_ros/CMakeLists.txt

```diff
@@ -1,245 +1,89 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.5)
 project(pcl_ros)
 
-## Find system dependencies
-find_package(cmake_modules REQUIRED)
-find_package(Boost REQUIRED COMPONENTS system filesystem thread)
-find_package(Eigen3 REQUIRED)
-find_package(PCL REQUIRED COMPONENTS core features filters io segmentation surface)
-
-if(NOT "${PCL_LIBRARIES}" STREQUAL "")
-  # For debian: https://github.com/ros-perception/perception_pcl/issues/139
-  list(REMOVE_ITEM PCL_LIBRARIES
-    "vtkGUISupportQt"
-    "vtkGUISupportQtOpenGL"
-    "vtkGUISupportQtSQL"
-    "vtkGUISupportQtWebkit"
-    "vtkViewsQt"
-    "vtkRenderingQt")
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
 endif()
-
-# There is a bug in the Ubuntu Artful (17.10) version of the VTK package,
-# where it includes /usr/include/*-linux-gnu/freetype2 in the include
-# directories (which doesn't exist).  This filters down to the PCL_INCLUDE_DIRS,
-# and causes downstream projects trying to use these libraries to fail to
-# configure properly.  Here we remove those bogus entries so that downstream
-# consumers of this package succeed.
-if(NOT "${PCL_INCLUDE_DIRS}" STREQUAL "")
-  foreach(item ${PCL_INCLUDE_DIRS})
-    string(REGEX MATCH "/usr/include/.*-linux-gnu/freetype2" item ${item})
-    if(item)
-      list(REMOVE_ITEM PCL_INCLUDE_DIRS ${item})
-    endif()
-  endforeach()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
 
-## Find catkin packages
-find_package(catkin REQUIRED COMPONENTS
-  dynamic_reconfigure
-  message_filters
-  nodelet
-  nodelet_topic_tools
+## Find system dependencies
+find_package(Eigen3 REQUIRED)
+find_package(PCL REQUIRED QUIET COMPONENTS core features filters io segmentation surface)
+
+## Find ROS package dependencies
+find_package(ament_cmake_ros REQUIRED)
+find_package(pcl_conversions REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(sensor_msgs REQUIRED)
+find_package(tf2 REQUIRED)
+find_package(tf2_geometry_msgs REQUIRED)
+find_package(tf2_ros REQUIRED)
+
+ament_export_dependencies(
```

Comment:
The ament_* macros are typically at the end of the files

Reply:
Usually we try to: `set(depedencies ...)` and then use `${dependencies}` in the export / linking so that we don't have multiple lists to edit to add a new dependency

Reply:
I opted against the global ${dependencies} because each sub-component is going to depend and link against very different things. It felt wrong to just broadly link against everything for each component (from the cmake/export perspective). But I guess it doesn't really matter.

Reply:
You could also make different dependency variables for different sets of components, but hardcoding `N` times `M` dependencies is going to be error prone to deal with if we add more.